### ISSUE TITLE
ALASQL substr support

### ISSUE THREAD

**Issue by vthilak** [agershun/alasql#393]
Hi, in order to incorporate a "substr" SQLite functionality into the alasql framework, would it be right to use 

https://github.com/agershun/alasql/wiki/User%20Defined%20Functions

Or is there a better way. 

Basically , when querying on a table with "date" column, we want to extract the "year" substr and use that for custom search .

**Comment by agershun** [comment]
1. To extract the year from the date (string or JavaScript) you can try to use YEAR() function:
```sql
    SELECT YEAR(GETDATE())
```
See the example [here](http://alasql.org/console/?SELECT%20YEAR(GETDATE()))

2. You also can use MID() function (or also it has the alias SUBSTRING() ) (this is standard substring function for many SQL dialects):

```sql
    SELECT MID("01-13-2015",7,4)
```
See the example [here](http://alasql.org/console/?SELECT%20MID(%2201-13-2015%22,7,4))

Here - the first character is 1.

3. Or you can use JavaScript ```.substr()``` function:
```sql
    SELECT "01-27-2016" -> substr(6,4)
```
See the example [here](http://alasql.org/console/?SELECT%20%2201-27-2016%22%20-%3E%20substr(6,4))

Here - the first leftmost character is 0.

4. Or you can define user-defined function:

```js
alasql.fn.SUBSTR = function(a,b,c) {
    return a.substr(b-1,c);
}
```

5. And the final option: We just add SUBSTR alias for MID() function, so you can download last version from the ```develop``` branch. The leftmost character is 1. :)

**Comment by mathiasrw** [comment]
alasql.fn.SUBSTR = function(a,b,c) {
        return a.substr(b-1,c);
    }

could be part of the consequenses of setting `config.dialect = 'sqlite'`

**Comment by agershun** [comment]
1. alasql.options.dialect

2. Yes, unfortunately, there are so many differences among dialects even for such simple function

Отправлено с iPhone

**Comment by mathiasrw** [comment]
Maybe the default (alasql) dialect will support terms from all the dialects that does not tamper with the same functionality.

**Comment by agershun** [comment]
Agree

2015-08-13 15:18 GMT+03:00 Mathias Rangel Wulff <notifications@github.com>:

**Comment by mathiasrw** [comment]
Closed per request from #628
